argocd/provider.go

```diff
@@ -269,3 +338,158 @@ func initApiClient(d *schema.ResourceData) (
 	}
 	return apiclient.NewClient(&opts)
 }
+
+func kubernetesResource() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"host": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("KUBE_HOST", ""),
```

Comment:
This is also copied from the Helm provider, so that it behaves the same way.

https://github.com/hashicorp/terraform-provider-helm/blob/main/helm/provider.go#L147

Reply:
I can add a comment on the source of this whole logic if you would like to.